user-defined-functions/specifications.yaml

```diff
@@ -0,0 +1,200 @@
+# Approach:
+# - Try to express patterns in terms of BNFC (https://bnfc.digitalgrammars.com/) (with slight modifications)
+# - Use <label :: (Type)> expressions to label non-terminals
+
+types:
+  - type: Binding
+    examples:
+      - and ↦ Φ.you
+patterns:
+  - pattern: |
+      MetaBinding. Binding ::= ((<topBindingName :: (MetaId)> "=>")? "\\" <binding :: (MetaId)>)? <path :: (MetaId)> ".." <name :: (MetaId)> "↦" <object :: (Object)>
+    description: |
+      A <path> through formations to <name> where a value after "↦" matches <object>.
+      If <topBindingName> is given, it's equal to the first element of <path>.
+      If <binding> is given, it's equal to the full binding.
+    examples:
+      - pattern: '\?PathHead => \?FullBinding ?Path..?Name ↦ ?Object.world(?Bindings)'
+        context: formation
+        input: "⟦ org ↦ ⟦ eolang ↦ ⟦ main ↦ hello.world(and ↦ Φ.you) ⟧ ⟧ ⟧"
+        matches:
+          - metavar: "?Path"
+            value: ["org", "eolang"]
+          - metavar: "?Name"
+            value: "main"
+          - metavar: "?Object"
+            value: "hello"
+          - metavar: "?Bindings"
+            value: [and ↦ Φ.you]
+          - metavar: "?PathHead"
+            value: ["org"]
+          - metavar: "?FullBinding"
+            value: [org ↦ ⟦ eolang ↦ ⟦ main ↦ hello.world(and ↦ Φ.you) ⟧ ⟧]
+      - pattern: "?Path..?Name ↦ ?DispatchPath..?DispatchName.world"
+        context: application
+        input: "(x ↦ org.eolang.world)"
+        matches:
+          - metavar: "?Path"
+            value: []
+          - metavar: "?Name"
+            value: "x"
+          - metavar: "?DispatchPath"
+            value: ["org"]
+          - metavar: "?DispatchName"
+            value: "eolang"
+  - pattern: |
+      ExpressionObject. Expression ::= Object
+    examples:
+      - ξ
+      - ⟦ a ↦ ?B ⟧
+  - pattern: |
+      ExpressionFunction. Expression ::= Function
+    examples:
+      - $name(?a)
+functions:
+  - pattern: |
+      FunctionMt. Function ::= "$mt" "(" <prefix :: (Expression)> "," <suffix :: (Expression)> ")"
+    description: |
+      Return the string "<prefix>.<suffix>".
```

Comment:
I do not think we should work with strings. We should work with AST. `<prefix>` should be a valid $\varphi$-term and `<suffix>` should be a valid attribute (or chain of dispatch/application).

Also naming is not explained. Why `mt`?

Reply:
And also, why do we need `$mt` if we can simply use dot `.`?

Reply:
> I do not think we should work with strings.

Yes, it should be a dispatch.

> Also naming is not explained. Why mt?

IDK. It comes from use cases. I'd prefer $dot.

>  And also, why do we need $mt if we can simply use dot?

I believe we shouldn't use dot with functions, should we?

1. `a.$name(b)`
1. `$dot(a, $name(b))`